app/models/Classroom.coffee

```diff
@@ -84,6 +86,8 @@ module.exports = class Classroom extends CocoModel
         continue
       levelObjects.push(course.levels)
     levels = new Levels(_.flatten(levelObjects))
+    language = @get('aceConfig')?.language
+    levels.remove(levels.filter((level) => level.get('primerLanguage') is language)) if language
```

Comment:
My first thought: consolidate the level-filtering into a method on the Levels collection, passing in the language from the classroom.

This makes it a bit inconsistent with the other filtering happening in classroom, though. I haven't quite worked out a good consistent way to put this kind of collection-specific filtering into the collection especially when it's dependent on outside information.

Maybe a class method on Levels, that you apply to a Levels collection? Or an instance method that mutatingly filters the collection it's called from?